src/dbnode/storage/series/buffer.go

```diff
@@ -369,7 +548,7 @@ func (b *dbBuffer) Snapshot(ctx context.Context, blockStart time.Time) (xio.Segm
 		err error
 	)
 
-	b.forEachBucketAsc(func(bucket *dbBufferBucket) {
+	b.forEachBucketAsc(bucketTypeAll, func(bucket *dbBufferBucket) {
```

Comment:
Do we want bucketTypeAll here? Seems like we'll need some kind of separate code path for snapshotting non-realtime blocks, otherwise the data will end up in the wrong fileset files

Reply:
I guess technically your code probably won't allow a realtime and non-realtime block for the same blockStart to exist at the same time so maybe it doesn't matter, but seems better to only go over realtime blocks here